Modules/SubModule_Import/xASL_wrp_DCM2NII_Subject.m

```diff
@@ -121,6 +121,14 @@
                 scanFields.iSession = iSession;
                 scanFields.iScan = iScan;
                 scanFields.name = thisRun.name;
+
+				% Create the session number based on the session name in format ASL_X
+				if isempty(thisRun.name) || isempty(regexpi(thisRun.name, '.*\d*$', 'once'))
+					% Set to 1 if session name cannot be identified
+ 					scanFields.numberSession = 1;
+				else
+					scanFields.numberSession = xASL_str2num(thisRun.name(5:end));
```

Comment:
This doesn't fully match with the regexp on line 126. So this should be using xASL_str2num(regexp(...)) instead (we strictly speaking don't know if it is always 5:end).

Alternatively, the first regexpi on line 126 should match '^ASL_\d*$

by the way, it should be \d+ not \d* (the asterisk also allows zero instances of the digit, which we don't want). So it should be '^ASL_\d+$' then right? Unless we want to be less restrictive, but then line 130 should change to a regular expression